### PR TITLE
Migrate to tar-0.6

### LINKED ISSUE #1277: Support tar-0.6
`hackage-server` demands `tar < 0.6`, prohibiting a recently released https://hackage.haskell.org/package/tar-0.6.0.0.

### PR DESCRIPTION
**Pull request by Bodigrim**
Closes #1277.

### CODE REVIEW

**Comment by Bodigrim** [comment]
Could someone help me out with Nix Flake please? Never used Nix in my life unfortunately.

**Comment by peterbecich** [comment]
Fixing it, will append a commit to this PR, sorry for the inconvenience

**Comment by peterbecich** [comment]
I think it is fixed. Pardon the `git rebase`. It was fixed by `nix flake update` and other tweaks to the Flake so `flake update` would succeed:  https://github.com/haskell/hackage-server/pull/1285 Your addition to the Flake `tar.source = "0.6.0"` was correct; the `flake` was too far behind NixPkgs for this to work, however.

**Comment by Bodigrim** [comment]
Thanks @peterbecich! 

@gbaz @ysangkok @andreasabel this is ready for review.

File: src/Distribution/Server/Packages/Unpack.hs [hunk 1]
```diff
@@ -468,6 +480,10 @@ explainTarError (PermissionsError entryname mode) =
   where
     showMode :: Tar.Permissions -> String
     showMode m = printf "%.3o" (fromIntegral m :: Int)
+explainTarError (LongNamesError err) =
+    "There is an error in the format of entries with long names in the tar file: " ++ show err
+ ++ ". Check that it is a valid tar file (e.g. 'tar -xtf thefile.tar'). "
+ ++ "You may need to re-create the package tarball and try again."
```
Comment (andreasabel) [inline src/Distribution/Server/Packages/Unpack.hs:486]: Can we get a test case that triggers this error?  

Could be not easy to engineer, but it is good to have evidence for every error that it is possible to trigger.

**Review by andreasabel — COMMENTED** [review]
Thanks for the upgrade!

File: src/Distribution/Server/Packages/Unpack.hs (hunk 1 above)
Comment (Bodigrim) [reply src/Distribution/Server/Packages/Unpack.hs:486]: Are there tests for pre-existing errors somewhere? I can add a new one there.

This is a rare condition though: one has to have an intent to craft a malicious TAR, you cannot obtain one by mistake.

Comment (andreasabel) [reply src/Distribution/Server/Packages/Unpack.hs:486]: > Are there tests for pre-existing errors somewhere? I can add a new one there.

Yeah, that is the usual problem: There aren't enough tests to start with, so there isn't a good pretext to add some for new or changed features.  One can't require adding a test infrastructure for a maintenance PR like this.

**Review by andreasabel — APPROVED** [review]

**Comment by Bodigrim** [comment]
@gbaz @ysangkok any more reviews? Or are you happy to merge?

**Comment by ysangkok** [comment]
Looks ok to me. Happy to merge.

**Pull request merged**